inference/core/workflows/core_steps/cache/cache_get/v1.py

```diff
@@ -0,0 +1,85 @@
+from typing import Any, List, Literal, Optional, Type, Union
+
+from pydantic import ConfigDict, Field
+
+from inference.core.workflows.core_steps.cache.memory_cache import WorkflowMemoryCache
+from inference.core.workflows.execution_engine.entities.base import (
+    OutputDefinition,
+    WorkflowImageData,
+)
+from inference.core.workflows.execution_engine.entities.types import (
+    IMAGE_KIND,
+    STRING_KIND,
+    WILDCARD_KIND,
+    Selector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+LONG_DESCRIPTION = """
+Fetches a previously stored value from a cache entry.
+
+Use the `Cache Set` block to store values in the cache.
+"""
+
+SHORT_DESCRIPTION = "Fetches a previously stored value from a cache entry."
+
+
+class BlockManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": "Cache Get",
+            "version": "v1",
+            "short_description": SHORT_DESCRIPTION,
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "fusion",
+        }
+    )
+    type: Literal["roboflow_core/cache_get@v1"]
+    image: Selector(kind=[IMAGE_KIND], pattern=r"(^\$inputs.[A-Za-z_0-9\-]+$)") = Field(
```

Comment:
is the reason for making pattern a parameter that you only want to allow input selectors?

We did a larger change a while ago to make the selectors compatible across input/step outputs and rely only on type instead.  Not sure this would be handled properly in the UI

Reply:
Yes, that's the goal (it's to force something we know should have video metadata if we're running on video).

I didn't notice it not working in the UI; if it isn't I can followup with a frontend PR. (There are some other "nice to have" things I noticed with some of these blocks as well that I was going to refine in the frontend.)